app/models/ext_management_system.rb

```diff
@@ -92,6 +92,12 @@ def hostname_uniqueness_valid?
     errors.add(:hostname, N_("has to be unique per provider type")) if existing_hostnames.include?(hostname.downcase)
   end
 
+  def hostname_format_valid?
+    return if hostname.ipaddress? || hostname.hostname?
+    error_msg = N_("is in a wrong format.")
+    errors.add(:hostname, _(error_msg))
```

Comment:
I think you want:
```ruby
errors.add(:hostname, _("format is invalid."))
```
Is that correct @mzazrivec ?
https://github.com/ManageIQ/guides/blob/master/i18n.md#ruby--haml

Reply:
Correct.